zed: yoga merge

markgoddard opened the pull request:
- CI: Update aio jobs to run on ARC
- Support credentials for custom DNF repositories
- CI: Use Ark package repositories to install packages
- CI: Switch from sms-lab-release to openstack cloud
- CI: Update aio VM flavor, network and subnet for Leafcloud
- CI: Update GitHub action versions for node20
- Revert hashicorp/setup-terraform action to v2 due to failure
- CI: Revert to ens3 interface for Leafcloud
- Add Apt auth configuration
- CI: Increase flavor to en1.large - some jobs were failing Tempest jobs
- CI: Run container image build job on ARC cluster
- Add package repository credentials to container image build
- Don't verify Apt repo CA initially when using HTTPS in container build

MaxBed4d committed:
CI: Update aio jobs to run on ARC

The jobs now run in containers, so some changes are necessary.

markgoddard committed:
CI: Run container image build job on ARC cluster

The Actions Runner Controller (ARC) cluster runs GitHub Actions jobs on
a Kubernetes cluster. This allows us to dynamically provision resources
for our CI workflows.

This change switches the container image build jobs to use the ARC
cluster. The job now runs in a container, with access to a Docker in
Docker container for building images. This allows us to skip the host
configuration part of the workflow.

See https://github.com/stackhpc/ARC-Installer/

markgoddard committed:
Support credentials for custom DNF repositories

This allows us to use packages from authenticated repositories such as
those on Ark. This patch depends on
https://review.opendev.org/c/openstack/kayobe/+/908142/

markgoddard committed:
CI: Use Ark package repositories to install packages

Previously we were using Test Pulp on SMS lab, but this is out of
action. Switching to Ark allows CI jobs to run on Leafcloud (or anywhere
with Internet access).

markgoddard committed:
CI: Switch from sms-lab-release to openstack cloud

This more generic name should be more portable.

markgoddard committed:
CI: Update aio VM flavor, network and subnet for Leafcloud

markgoddard committed:
CI: Update GitHub action versions for node20

node16 support is deprecated and will be dropped.

markgoddard committed:
CI: Revert to ens3 interface for Leafcloud

Leafcloud is not using the q35 machine type yet.

markgoddard committed:
Add package repository credentials to container image build

This is necessary to access package repositories hosted on Ark.

markgoddard committed:
Add Apt auth configuration

This allows us to use packages from authenticated repositories such as
those on Ark. This patch depends on
https://review.opendev.org/c/openstack/kayobe/+/908413

markgoddard committed:
CI: Increase flavor to en1.large - some jobs were failing Tempest jobs

markgoddard committed:
Revert hashicorp/setup-terraform action to v2 due to failure

markgoddard committed:
Don't verify Apt repo CA initially when using HTTPS in container build

The Ubuntu base image doesn't contain the ca-certificates package, so we
can't verify an HTTPS package mirror to download the ca-certificates
package. Using the upstream repos may result in a version conflict when
we switch to another mirror.

markgoddard committed:
Merge pull request #917 from stackhpc/CleanAiOChanges

CI: Use ARC runners for aio jobs, move CI to Leafcloud

markgoddard committed:
Merge pull request #922 from stackhpc/yoga-container-build-on-arc

CI: Use ARC runners for container image build jobs

markgoddard committed:
Merge stackhpc/yoga into stackhpc/zed

markgoddard committed:
CI: Fix up check-tags job for ARC

markgoddard committed:
CI: Trim down package dependencies

Alex-Welsh commented on .github/workflows/stackhpc-check-tags.yml:
```diff
@@ -20,14 +20,15 @@ jobs:
       KAYOBE_IMAGE: ${{ inputs.kayobe_image }}
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout
```
Might need to install/update git before this. https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/7842242364/job/21400456106?pr=924#step:2:28

markgoddard replied:
Done

Alex-Welsh commented:
There I was thinking our days of TF errors were over :disappointed:

markgoddard committed:
CI: Add retries to TF apply in aio job

We sometimes hit temporary failures during VM creation. Adding a retry
may help to keep things running.

markgoddard committed:
CI: Fix up aio upgrade job for ARC runner

This job is not present in yoga, so was not tested there.

markgoddard commented:
This is mostly working now, just waiting on upgrade jobs. The check container image tags job needs some fixes because the stackhpc-dev container repositories on Ark are "push repositories". It's non-voting though, so I suggest we merge and fix that later.

Alex-Welsh commented on .github/workflows/stackhpc-all-in-one.yml:
```diff
@@ -168,7 +177,17 @@ jobs:
 
       - name: Terraform Apply
-        run: terraform apply -auto-approve
+        run: |
```
Since this is just a command, we could try this retry action:
https://github.com/nick-fields/retry

Alex-Welsh replied:
Just to clarify, I'm happy to merge as is. I don't think there's a huge benefit to using the custom action, but worth considering as an option.

markgoddard replied:
I'm generally a bit wary of unofficial custom actions because they can have access to some sensitive data.

Alex-Welsh commented on .github/workflows/stackhpc-container-image-build.yml:
```diff
@@ -132,23 +127,19 @@ jobs:
+      - name: Configure localhost as a seed
         run: |
-          sudo docker image prune --all --force --filter until=168h --filter="label=kolla_version"
+          cat > src/kayobe-config/etc/kayobe/environments/ci-builder/inventory/hosts << EOF
```
Is this not something we can do in SKC permanently?

markgoddard replied:
Possibly eventually. Currently the host image build workflow doesn't work in ARC, so we'd need to use the old style workflow which hasn't been tested with this.

Alex-Welsh commented on etc/kayobe/kolla.yml:
```diff
@@ -250,12 +250,26 @@ kolla_build_blocks:
                -e '/\[{{ repo.tag }}\]/,/^\[/ s/^[# ]*\(metalink *=.*\)/#\1/g' \
+    {% if stackhpc_repo_mirror_username is truthy %}
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1\nusername={{ stackhpc_repo_mirror_username }}|' \
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1\npassword={{ stackhpc_repo_mirror_password }}|' \
```
I'm not sure if there's a good way around it, but this could create some hard-to-debug errors if the password contains a `|`

markgoddard replied:
We could escape it, but a fix should go to stackhpc/yoga. @jovial had some valid concerns around the passwords being in lower layers of the images, even though we remove them, so we may need to revisit this.

Alex-Welsh reviewed (approved):
Couple of threads to follow up on but LGTM as a fix for our AIO issues
